### PR TITLE
feat: dedupe on commit

### PR DESCRIPTION
**Pull request by JFrankfurt**
## Description
Adds an automatic hook to run `yarn yarn-deduplicate --strategy=highest` whenever yarn.lock changes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,16 @@
...
     "postinstall": "yarn patch-package"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "yarn.lock": [
+      "yarn yarn-deduplicate --strategy=highest"
```
Comment (zzmp) [inline package.json:43]: why not just run `yarn deduplicate`, and use our existing script? This seems more prone to skew.

File: package.json [hunk 2]
```diff
@@ -33,6 +33,16 @@
     "deduplicate": "yarn-deduplicate --strategy=highest",
     "postinstall": "yarn patch-package"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
```
Comment (zzmp) [inline package.json:38]: won't this cost some time?

**Review by zzmp — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (JFrankfurt) [reply package.json:38]: yeah, but less than 1 second

File: package.json (hunk 1 above)
Comment (JFrankfurt) [reply package.json:43]: did not know this existed, thanks

**Review by zzmp — APPROVED** [review]

**Review by JackShort — APPROVED** [review]

**Pull request merged**